runtime-sdk/src/dispatcher.rs

```diff
@@ -346,10 +373,25 @@ impl<R: Runtime> transaction::dispatcher::Dispatcher for Dispatcher<R> {
         // TODO: Get rid of StorageContext (pass mkvs in ctx).
         StorageContext::with_current(|mkvs, _| {
             // Prepare dispatch context.
-            let mut ctx = DispatchContext::from_runtime(&ctx, mkvs, &self.methods);
+            let mut ctx = DispatchContext::from_runtime(&ctx, mkvs);
             // Perform state migrations if required.
             self.maybe_init_state(&mut ctx);
 
+            match method {
+                "core.EstimateGas" => {
```

Comment:
note: our linter doesn't like this single-case match. but it feels so much like a match

also side note: I previously tried adding a method to the method registry for this, but the registry is built for static functions, where as this implementation accesses the dispatcher instance. and rust probably wouldn't be a good sport about having a self-referential closure.